doc/reference/trd104-syscalls.md

```diff
@@ -193,35 +213,37 @@ are additional error codes to include errors related to userspace.
 | 8     | CANCEL      | The operation was actively cancelled by a call to a cancel() method or function.        |
 | 9     | NOMEM       | The operation required memory that was not available (e.g. a grant region or a buffer). |
-| 10    | NOSUPPORT   | The operation is not supported/implemented.                                             |
-| 11    | NODEVICE    | The specified device is not implemented by the kernel.                                  |
+| 10    | NOSUPPORT   | The system call is not available to or not supported for the calling process.           |
+| 11    | NODEVICE    | The driver specified by the driver identifier is not available to the calling process.  |
 | 12    | UNINSTALLED | The resource was removed or uninstalled (e.g., an SD card).                             |
 | 13    | NOACK       | The packet transmission was sent but not acknowledged.                                  |
 | 1024  | BADRVAL     | The variant of the return value did not match what the system call should return.       |
 
-Values in the range 1-1023 reflect kernel return value error codes. Kernel error
-codes not specified above are currently reserved. TRDs MAY specify reserved
-kernel error codes, but MUST NOT specify kernel error codes greater than 1023.
-The Tock kernel MUST NOT return an error code unless the error code is specified
-in a TRD.
+Values in the range 1-1023 reflect kernel return value error
+codes. Kernel error codes not specified above are reserved. TRDs MAY
+specify additional kernel error codes from these reserved values, but
+MUST NOT specify kernel error codes greater than 1023.  The Tock
+kernel MUST NOT return an error code unless the error code is
+specified in a TRD.
 
 Values greater than 1023 are reserved for userspace library use. Value 1024
 (BADRVAL) is for when a system call returns a different failure or success
-variant than the userspace library expects.
-
+variant than the userspace library expects. 
 
 4 System Call API
 =================================
 
-Tock has 7 classes or types of system calls. When a system call is invoked, the
-class is encoded as the Syscall Class ID. Some system call classes are implemented
-by the core kernel and so always have the same operations. Others are implemented
-by peripheral syscall drivers and so the set of valid operations depends on what peripherals
-the platform has and which have drivers installed in the kernel.
+Tock has 7 classes or types of system calls. When a system call is
+invoked, the class is encoded as the Syscall Class ID. Some system
+call classes are implemented by the core kernel and so the supported
+calls are the same across kernels. Others are implemented by system
+call drivers, which can be added and removed in different kernel
+builds. The full set of valid system calls a kernel supports therefore
+depends on what system call drivers it has installed.
 
 The 6 classes are:
```

Comment:
Right above the table it says "The 6 classes are:" when there are 7 classes.

Reply:
Fixed